vernac/g_vernac.mlg

```diff
@@ -744,18 +744,19 @@ GRAMMAR EXTEND Gram
           { VernacCanonical CAst.(make ~loc @@ ByNotation ntn) }
 
       (* Coercions *)
-      | IDENT "Coercion"; qid = global; u = OPT univ_decl; d = def_body ->
-          { let s = coerce_reference_to_id qid in
-          VernacDefinition ((NoDischarge,Coercion),((CAst.make ?loc:qid.CAst.loc (Name s)),u),d) }
+      | IDENT "Coercion"; qid = global; ud = OPT [ u = OPT univ_decl; d = def_body -> { u, d } ] ->
```

Comment:
Indeed, I was unable to improve the actual grammar as I don't master the constraints of the parsing engine.
I thus used `common.edit_mlg` as you suggested to make it look better in the refman (while hopefully being equivalent).

Reply:
Actually, my suggestion was wrong, sorry--the change would allow a univ_decl without a def_body while the original would not.  You'll want to undo some of the `common.edit_mlg` change.

Reply:
No worries, that's one of the reason why I edited `common.edit_mlg` differently. There should be no need to undo anything.